jira-config/jira-config-platform/src/main/java/org/xwiki/contrib/jira/config/internal/JIRAConfigurationMigrator.java

```diff
@@ -0,0 +1,184 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.jira.config.internal;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.xwiki.bridge.event.ApplicationReadyEvent;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.job.event.JobFinishedEvent;
+import org.xwiki.model.document.DocumentAuthors;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.model.reference.LocalDocumentReference;
+import org.xwiki.model.reference.WikiReference;
+import org.xwiki.observation.AbstractEventListener;
+import org.xwiki.observation.event.Event;
+import org.xwiki.query.QueryManager;
+import org.xwiki.user.UserReference;
+import org.xwiki.user.internal.document.DocumentUserReference;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * Listener to handle migration of authentication in JIRA configuration.
+ *
+ * @version $Id$
+ * @since 11.0.0
+ */
+@Singleton
+@Component
+@Named(JIRAConfigurationMigrator.HINT)
+public class JIRAConfigurationMigrator extends AbstractEventListener
+{
+    /**
+     * Main name of the listener.
+     */
+    public static final String HINT = "JIRAConfigMigrationListener";
+
+    private static final String JIRA = "JIRA";
+
+    private static final String PROPERTY_USERNAME = "username";
+
+    private static final String PROPERTY_PASSWORD = "password";
+
+    private static final String PROPERTY_ID = "id";
+
+    private static final String PROPERTY_AUTHENTICATION_TYPE = "authenticationType";
+
+    private static final String DOC_SAVE_MESSAGE = "Migrate configration";
+
+    private static final LocalDocumentReference MIGRATION_STATUS_DOCUMENT_REFERENCE =
+        new LocalDocumentReference(JIRA, "JIRAConfigAuthMigration");
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private Provider<QueryManager> queryManagerProvider;
+
+    @Inject
+    @Named("currentmixed")
+    private DocumentReferenceResolver<String> currentMixedReferenceResolver;
+
+    @Inject
+    private Logger logger;
+
+    /**
+     * Public constructor.
+     */
+    public JIRAConfigurationMigrator()
+    {
+        super(HINT, List.of(new ApplicationReadyEvent(), new JobFinishedEvent("install")));
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        migrate();
```

Comment:
When you receive JobFinishedEvent the context wiki will always be the main wiki, but the upgrade might have been actually been done on a subwiki and I assume you need to do the migration in a subwiki in that case.

Reply:
IMO you need a new method used as an oracle to know if migrate should be called or not: right now this listener will be called each time the wiki is restarted and will call migrate() which doesn't perform any check.